### PR TITLE
[WIP] Convert to v1alpha2

### PR DESCRIPTION
**Pull request by chuckha**
**What this PR does / why we need it**:
This PR sets us up for v1alpha2 implementation by getting all the types in place and go.mod happy.

Things that will need to happen:

## Actuator/controller

The actuator is going away in favor of the DockerMachineController. I left the logic in the actuators so we could copy/pasta or use as reference.

## Capdctl 

Updates to capdctl will be extensive. It needs to set up a management cluster which will include fetching and setting up the infra provider manager, the bootstrap provider manager and capi manager.

It needs to add all the v1a2 CRDs from all providers.

# CRDs

This change introduces the DockerMachine crd in the `infrastructure.x-k8s.io` group. It would be great to flesh this out and keep things like the version of the kind node inside, but that's TBD. Definitely up for discussion here

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Related to #54 

**Special notes for your reviewer**:

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: hack/boilerplate.go.txt [hunk 1]
```diff
@@ -0,0 +1,14 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (chuckha) [inline hack/boilerplate.go.txt]: missing copyright

**Comment by amy** [comment]
Do we need the docker provider to also be able to support v1alpha1?

**Comment by chuckha** [comment]
> Do we need the docker provider to also be able to support v1alpha1?

Yep, see #54 for details.

**Comment by chuckha** [comment]
Split this into 4 PRs. Closing this one

/close

**Pull request closed without merging**